From: Jiang <jiangxin@ossxp.com>
Subject: [PATCH] t/bugfix_loale_cannot_save

Testlink 1.8.5b introduces a new bug: if user change profile,
locale setting will lost, and can not restore any more.

Signed-off-by: Jiang <jiangxin@ossxp.com>

---
 lib/usermanagement/userInfo.php |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lib/usermanagement/userInfo.php b/lib/usermanagement/userInfo.php
index 1480069..b7ac452 100644
--- a/lib/usermanagement/userInfo.php
+++ b/lib/usermanagement/userInfo.php
@@ -101,7 +101,7 @@ function init_args()
     $args->id = isset($_REQUEST['id']) ? intval($_REQUEST['id']) : 0;
 
     $args->user = new stdClass();
-	$args->user->locale = isset($_REQUEST['id']) ? substr($args->operation,0,10) : 'en_GB';
+	$args->user->locale = isset($_REQUEST['id']) && isset($_REQUEST['locale']) ? substr(trim($_REQUEST['locale']),0,10) : 'en_GB';
     $key2loop = array('firstName','lastName','emailAddress');
 	foreach($key2loop as $key)
     {
-- 
tg: (f865ee3..) t/bugfix_loale_cannot_save (depends on: master)
